### ISSUE TITLE
Getting `unspecified-encoding` when using Path().open('rb')

### ISSUE THREAD

**Issue by ecs-jnguyen** [PyCQA/pylint#5018]
### Bug description

I am running into the error `unspecified-encoding` when using `Path().open("rb")`

Example python code:

file.py

```python
"""Module docstring."""
from pathlib import Path

hello_txt = Path("hello.txt")

# file.py:7:5: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
with hello_txt.open("rb") as stream:
    print(stream.read().decode())
```

hello.txt

```txt
hello there
```

### Command used

```shell
pylint file.py
```

### Pylint output

```shell
************* Module file
file.py:7:5: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)
```

### Expected behavior

We should not see the unspecified-encoding error because we are opening the file in binary mode. When I try to open the file with `with hello_txt.open("rb", encoding="utf-8") as stream:` I will get the following output

```
Traceback (most recent call last):
  File ".../file.py", line 7, in <module>
    with hello_txt.open("rb", encoding="utf-8") as stream:
  File ".../3.9.6/lib/python3.9/pathlib.py", line 1242, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
ValueError: binary mode doesn't take an encoding argument
```

### Pylint version

```shell
pylint 2.11.0
astroid 2.8.0
Python 3.9.6 (default, Sep  1 2021, 13:57:59) 
[Clang 12.0.0 (clang-1200.0.32.28)]
```

### OS / Environment

Mac and iterm2

### Additional dependencies

astroid==2.8.0
isort==5.9.3
lazy-object-proxy==1.6.0
mccabe==0.6.1
platformdirs==2.3.0
pylint==2.11.0
toml==0.10.2
typing-extensions==3.10.0.2
wrapt==1.12.1

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue, This is a duplicate of #5017 which was open in parallel.

**Comment by ecs-jnguyen** [comment]
ah i was beat by less than 15 minutes haha
